nixos/modules/services/web-apps/funkwhale.nix

```
@@ -0,0 +1,543 @@
+          };
+
+          socket = mkOption {
+            type = types.nullOr types.path;
+            default = "/run/postgresql";
+            defaultText = "/run/postgresql";
+            example = "/run/postgresql";
+            description = "Path to the unix socket file to use for authentication for local connections.";
+          };
+
+          createLocally = mkOption {
+            type = types.bool;
+            default = true;
+            description = "Create the database and database user locally.";
+          };
+        };
+
+        dataDir = mkOption {
+          type = types.str;
+          default = "/srv/funkwhale";
+          description = ''
+            Where to keep the funkwhale data.
+          '';
+        };
+
+        apiIp = mkOption {
+          type = types.str;
+          default = "127.0.0.1";
+          description = ''
+            Funkwhale API IP.
+          '';
+        };
+
+        webWorkers = mkOption {
+          type = types.int;
+          default = 1;
+          description = ''
+            Funkwhale number of web workers.
+          '';
+        };
+
+        apiPort = mkOption {
+          type = types.port;
+          default = 5000;
+          description = ''
+            Funkwhale API Port.
+          '';
+        };
+
+        hostname = mkOption {
+          type = types.str;
+          description = ''
+            The definitive, public domain you will use for your instance.
+          '';
+          example = "funkwhale.yourdomain.net";
+        };
+
+        protocol = mkOption {
+          type = types.enum [ "http" "https" ];
+          default = "https";
+          description = ''
+            Web server protocol.
+          '';
+        };
+
+        emailConfig = mkOption {
+          type = types.str;
+          default = "consolemail://";
+          description = ''
+            Configure email sending. By default, it outputs emails to console instead of sending them. See https://docs.funkwhale.audio/configuration.html#email-config for details.
+          '';
+          example = "smtp+ssl://user@:password@youremail.host:465";
+        };
+
+        defaultFromEmail = mkOption {
+          type = types.str;
+          description = ''
+            The email address to use to send system emails.
+          '';
+          example = "funkwhale@yourdomain.net";
+        };
+
+        api = {
+          mediaRoot = mkOption {
+            type = types.str;
+            default = "/srv/funkwhale/media";
+            description = ''
+              Where media files (such as album covers or audio tracks) should be stored on your system ? Ensure this directory actually exists.
+            '';
+          };
+
+          staticRoot = mkOption {
+            type = types.str;
+            default = "/srv/funkwhale/static";
+            description = ''
+              Where static files (such as API css or icons) should be compiled on your system ? Ensure this directory actually exists.
```

```suggestion
              Where static files (such as API css or icons) should be compiled on your system. Ensure this directory actually exists.
```

Same for other descriptions where there's a random question mark.